### PR TITLE
build: Add dependency on `eslint-plugin-jsdoc`

### LINKED ISSUE #10317: Add `eslint-plugin-jsdoc` to allow enforcement of source code docs style, and potentially enforce docs presence in the future
## Feature
See [eslint-plugin-jsdoc](https://www.npmjs.com/package/eslint-plugin-jsdoc) for a more detailed overview of the plugin.

I see two primary benefits to this plugin:
1. Enforcing consistent source-code documentation style (correct argument names for functions, consistent ordering of tags, etc.)
2. Enforcing the presence of code docs on certain API elements
    - Note that this can be scoped in fairly fine-grained manner. If we only wish to enforce for certain types of API members, or only wish to enforce for `exported` members, etc., that can be done. (And, of course, we can always use these controls to roll out requirements gradually as desired).

I've created this initial issue to track adding the plugin itself as a dev dependency. Enforcement of any rules will be tracked separately.

### PR DESCRIPTION
**Pull request by Josmithr**
## [10317 - Add eslint-plugin-jsdoc to allow enforcement of source code docs style, and potentially enforce docs presence in the future](https://github.com/microsoft/FluidFramework/issues/10317)

For more details on the package, see [here](https://www.npmjs.com/package/eslint-plugin-jsdoc).

This PR adds a dependency on the package so we can utilize its rules in the future.

### CODE REVIEW

File: common/build/eslint-config-fluid/package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "eslint-plugin-editorconfig": "~3.2.0",
     "eslint-plugin-eslint-comments": "~3.2.0",
     "eslint-plugin-import": "~2.25.4",
+    "eslint-plugin-jsdoc": "~39.3.0",
```
Comment (tylerbutler) [inline common/build/eslint-config-fluid/package.json:29]: You won't be able to apply any fixes without also adding this to the individual packages, too. Might as well do it here to keep the code change PR really focused on just code changes.

Comment (Josmithr) [reply common/build/eslint-config-fluid/package.json:29]: Is that preferable over just adding the dependency in the individual packages as they turn on rules? I can see arguments for both options, so I am happy to take whatever approach we prefer, just wanted to clarify :)

Comment (tylerbutler) [reply common/build/eslint-config-fluid/package.json:29]: From a maintenance standpoint it's easier if all the packages have the peer dependencies that they need. While we don't express it in package.json, the eslint config technically has a peer dependency on the plugins that it uses, so it's also a bit more correct.

For making the change, I usually go low-tech. I would just use VSCode to search for `"eslint-plugin-import": "~2.25.4",` in package.json and replace it with `"eslint-plugin-import": "~2.25.4","eslint-plugin-jsdoc": "~39.3.0",`, then run policy-check:fix to clean up all the package.jsons. Then update the lockfiles.

Comment (Josmithr) [reply common/build/eslint-config-fluid/package.json:29]: Actually, I'm a bit confused here. The other packages should get the new dep automatically as a transitive dependency through the config package right? Why do they need to depend directly on the plugin?

Comment (Josmithr) [reply common/build/eslint-config-fluid/package.json:29]: Hmm, I could have sworn the transitive dep was sufficient even for plugins. We had a similar pattern in the Whiteboard framework repo, but the plugin packages didn't leak out like this. Is this just a local development issue due to the way the symlinks work out?

Comment (Josmithr) [reply common/build/eslint-config-fluid/package.json:29]: Happy to make the change here until we resolve this, but I think we can clean this up.

Comment (tylerbutler) [reply common/build/eslint-config-fluid/package.json:29]: There is a workaround that I attempted to integrate in #9562. When I went to apply the changes to downstream packages, though, I ran into problems when using fluid-build that caused a number of rules to throw exceptions, most notably unbound-method. I never had a chance to chase down the problem, but I think all the pieces are in place already to address it.

**Review by tylerbutler — APPROVED** [review]

File: common/build/eslint-config-fluid/package.json (hunk 1 above)
Comment (Josmithr) [reply common/build/eslint-config-fluid/package.json:29]: I created #10410 which demonstrates another option for resolving this.

**Pull request merged**